starknet_devnet/general_config.py

```diff
@@ -1,37 +1,42 @@
 """
 Contains general_config generation functionalities.
 """
-
 from starkware.starknet.definitions import constants
 from starkware.starknet.definitions.general_config import (
-    DEFAULT_CHAIN_ID,
     DEFAULT_GAS_PRICE,
     DEFAULT_MAX_STEPS,
     DEFAULT_SEQUENCER_ADDRESS,
     DEFAULT_VALIDATE_MAX_STEPS,
+    StarknetChainId,
     build_general_config,
 )
 
 from .constants import SUPPORTED_TX_VERSION
 from .fee_token import FeeToken
 
-DEFAULT_GENERAL_CONFIG = build_general_config(
-    {
-        "cairo_resource_fee_weights": {
-            "n_steps": constants.N_STEPS_FEE_WEIGHT,
-        },
-        "contract_storage_commitment_tree_height": constants.CONTRACT_STATES_COMMITMENT_TREE_HEIGHT,
-        "event_commitment_tree_height": constants.EVENT_COMMITMENT_TREE_HEIGHT,
-        "global_state_commitment_tree_height": constants.CONTRACT_ADDRESS_BITS,
-        "invoke_tx_max_n_steps": DEFAULT_MAX_STEPS,
-        "min_gas_price": DEFAULT_GAS_PRICE,
-        "sequencer_address": hex(DEFAULT_SEQUENCER_ADDRESS),
-        "starknet_os_config": {
-            "chain_id": DEFAULT_CHAIN_ID.name,
-            "fee_token_address": hex(FeeToken.ADDRESS),
-        },
-        "tx_version": SUPPORTED_TX_VERSION,
-        "tx_commitment_tree_height": constants.TRANSACTION_COMMITMENT_TREE_HEIGHT,
-        "validate_max_n_steps": DEFAULT_VALIDATE_MAX_STEPS,
-    }
-)
+
+def build_devnet_general_config(chain_id):
```

Comment:
`chain_id` should have a type hint. According to the way it's currently used, it would be a `: str`, but I would suggest to use `StarknetChainId`, and the actual the getting of value from the chain_id key is done in the validation function in devnet_config.py

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/358/commits/23bfccda0ad5ba80d28fac5ea852f40898b608ff

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/358/commits/4867352176e93953300c3a7e72c95a54980cce4f

I added `StarknetChainId` hint in `build_devnet_general_config` but I left `str` name value in `devnet_config.py`